common/test_filestore_utils.py

```diff
@@ -21,27 +21,76 @@
 # TODO(zhichengcai): Figure out how we can test filestore_utils when using
 # the local_filestore implementation.
 
+#pylint: disable=invalid-name,unused-argument
 
-def test_keyword_args():
-    """Tests that keyword args, and in particular 'parallel' are handled
-    correctly."""
-    filestore_path = 'gs://fake_dir'
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        filestore_utils.rm(filestore_path, recursive=True, parallel=True)
-        mocked_execute.assert_called_with(
-            ['gsutil', '-m', 'rm', '-r', filestore_path], expect_zero=True)
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        mocked_execute.return_value = new_process.ProcessResult(0, '', '')
-        filestore_utils.ls(filestore_path)
-        mocked_execute.assert_called_with(['gsutil', 'ls', filestore_path],
-                                          expect_zero=True)
-
-    filestore_path2 = filestore_path + '2'
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        filestore_utils.cp(filestore_path, filestore_path2, parallel=True)
-        mocked_execute.assert_called_with(
-            ['gsutil', '-m', 'cp', filestore_path, filestore_path2],
-            expect_zero=True)
+
+class TestLocalFilestore:
+    """Tests for switching on local_filestore."""
+    FILESTORE = '/experiment_data'
+    DIR1 = '/dir1'
+    DIR2 = '/dir2'
+
+    def test_local_filestore_on(self, switch_on_local_filestore):
+        """Tests that local_filestore switches on correctly."""
+        with mock.patch('common.new_process.execute') as mocked_execute:
+            filestore_utils.cp(self.DIR1, self.DIR2, recursive=True)
+            mocked_execute.assert_called_with(
+                ['cp', '-r', self.DIR1, self.DIR2], expect_zero=True)
```

Comment:
I would just do `assert 'gsutil' not in mocked_execute.call_args_list

Reply:
Good suggestion! Let me change to it. That is more related to the testing goal.